Tools/AP_Periph/battery_bms.cpp

```diff
@@ -0,0 +1,454 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  battery BMS includes a button which, when pressed, shows the state of charge percentage using LEDs
+ */
+#include "AP_Periph.h"
+
+#if AP_PERIPH_BATTERY_BMS_ENABLED
+#include "stdio.h"
+#include "battery_bms.h"
+
+extern const AP_HAL::HAL &hal;
+extern AP_Periph_FW periph;
+
+// configure gpio pins. returns true once configured
+bool BatteryBMS::configured()
+{
+    if (config_complete) {
+        return true;
+    }
+
+#if HAL_GPIO_LED_ON != 0
+    hal.gpio->pinMode(HAL_GPIO_PIN_BMS_LED1, HAL_GPIO_OUTPUT);
```

Comment:
Consider what we do elsewhere with this sort of structure - we essentially end up initialising a structure from a comma-separated list in the hwdef.  You then iterate of the length of that structure here rather than assuming 8 LEDs.

e.g. `define AP_PERIPH_BMS_LEDS 45,46,47,48,49, 50` in the hwdef and 

```
const uint8_t bms_leds[] { AP_PERIPH_BMS_LEDS };
```
here